src/github.com/getlantern/peerscanner/cf/cf.go

```diff
@@ -53,31 +54,66 @@ func (util *Util) GetAllRecords() ([]cloudflare.Record, error) {
 	return allRecords, nil
 }
 
-func (util *Util) Register(name string, ip string) (*cloudflare.Record, error) {
-	cr := cloudflare.CreateRecord{Type: "A", Name: name, Content: ip}
-	rec, err := util.Client.CreateRecord(util.domain, &cr)
+// Register ensures that a record with the given name and ip is registered and
+// proxying (orange cloud enabled). An existing record can optionally be passed
+// in, in which case the record is assumed to be registered and we enable the
+// orange cloud for the existing record.
+// EnsureRegistered returns:
+//  - the record if registration was successful
+//  - true of it was able to turn on proxying
+//  - any error encountered
+func (util *Util) EnsureRegistered(name string, ip string, rec *cloudflare.Record) (*cloudflare.Record, bool, error) {
+	if rec == nil {
+		// Register record
+		var err error
+		cr := cloudflare.CreateRecord{Type: "A", Name: name, Content: ip}
+		rec, err = util.Client.CreateRecord(util.domain, &cr)
 
-	if err != nil {
-		return nil, err
+		if err != nil {
+			if !isDuplicateError(err) {
+				return nil, false, err
+			}
+			log.Debugf("%v (%v) already registered, looking up existing record", name, ip)
+			// Note - this is pretty heavyweight since it fetches all
+			// records, but this condition should rarely be hit anyway
+			all, err := util.GetAllRecords()
+			if err != nil {
+				return nil, false, err
```

Comment:
Maybe log this? Do we have any way of seeing error logs on this guy in general? How easy would it be to link it up with Papertrail if we don't currently have a way to see 'em?